### ISSUE TITLE
🐛 Using `navigate = true` causes `git diff` with an empty diff to print `Pattern not found  (press RETURN)`

### ISSUE THREAD

**Issue by infokiller** [dandavison/delta#310]
Running `git diff` on `.bashrc` when it doesn't have changes:

![image](https://user-images.githubusercontent.com/10516814/91993723-6535c780-ed3e-11ea-9470-f98a9ae4828b.png)

**Comment by dandavison** [comment]
Hi @infokiller this looks like a duplicate of https://github.com/dandavison/delta/issues/295 and https://github.com/dandavison/delta/issues/237. The best workaround I have so far is https://github.com/dandavison/delta/issues/295#issuecomment-675531472 (it is also [mentioned](https://github.com/dandavison/delta#navigation-keybindings-for-large-diffs) in the README).

**Comment by infokiller** [comment]
Thanks @dandavison you're right, it's indeed a duplicate, sorry for missing the other issues.
It seems I should set navigate to false by default and use it only on demand.
BTW, any idea why `git -c delta.navigate=true diff` doesn't work (when the default in the config is set to false)?
I prefer something like this over mutating the global config with `git config --global` every time I want to enable this feature.

**Comment by dandavison** [comment]
> BTW, any idea why `git -c delta.navigate=true diff` doesn't work (when the default in the config is set to false)?

It's because `git -c ...` mutates the git config setting in the memory of the git process; but delta is reading git config settings from disk.

Btw there's some discussion of this here also: https://github.com/dandavison/delta/issues/288#issuecomment-673555456

**Comment by infokiller** [comment]
Hmm, in that case something like this works, in case anyone else wants to avoid mutating the git config:

```sh
_git_setenv_global_config_copy() {
  local tmpdir="${TMPDIR:-/tmp}/git-config-tmp"
  mkdir -p -- "${tmpdir}"
  local xdg_config_home="${xdg_config_home:-${HOME}/.config}"
  local fake_xdg_config_home
  fake_xdg_config_home="$(mktemp -d --tmpdir="${tmpdir}" "$$.XXXXX")"
  cp -R -- "${xdg_config_home}/git" "${fake_xdg_config_home}/git"
  while IFS='' read -r file; do
    ln -s "${xdg_config_home}/${file}" "${fake_xdg_config_home}/${file}"
  done < <(find "${xdg_config_home}" -mindepth 1 -maxdepth 1 -not -name git \
    -exec basename '{}' \;)
  export XDG_CONFIG_HOME="${fake_xdg_config_home}"
}

git-delta-navigate() {
  (
    _git_setenv_global_config_copy
    git config --global delta.navigate true
    git "$@"
  )
}
```

**Comment by dandavison** [comment]
Cool. I'm thinking that there's an argument for allowing navigate to be activated via an environment variable `DELTA_NAVIGATE`. Would you find that convenient?

**Comment by dandavison** [comment]
I actually went ahead and made that change and it is now released (0.4.3) i.e. the latest release of delta allows `navigate` to be activated via an environment variable, e.g.:

```
DELTA_NAVIGATE=1 git diff
```

(see the README section https://github.com/dandavison/delta#navigation-keybindings-for-large-diffs)

**Comment by infokiller** [comment]
Thanks @dandavison using an env var is better than my awkward workaround
